callbacks.py

```diff
@@ -115,34 +117,28 @@ async def audio(self, room, event):
         Arguments:
         ---------
             room (nio.rooms.MatrixRoom): The room the event came from
-            event (nio.events.room_events.RoomMessageAudio): The event
+            event (nio.events.room_events.RoomMessageAudio|nio.events.room_events.RoomEncryptedAudio): The event
                 defining the (audio) message
 
         """
         # Ignore messages from ourselves
         if event.sender == self.client.user:
             return
 
-        # Ignore non-ogg audio
-        if event.body[-4:] != '.ogg':
-            logger.debug(
-                f"Bot received (apparently) non-ogg data for room {room.display_name} | "
-                f"{room.user_name(event.sender)}: {event.body}"
-            )
-            return
-
         # download the audio data
         logger.debug(
-            f"Bot downloading audio for room {room.display_name} | "
+            f"Bot downloading audio data for message from room '{room.display_name}' | "
             f"{room.user_name(event.sender)}: {event.url}"
         )
         mxc = urlparse(event.url)
         response = await self.client.download(server_name=mxc.netloc, media_id=mxc.path.strip("/"), filename=None, allow_remote=True)
         if isinstance(response, DownloadError):
             logger.error(f"Bot download of media resulted in error")
             return
-        data = decrypt_attachment(response.body, event.key["k"], event.hashes["sha256"], event.iv)
-        data = f"data:audio/ogg;base64,{b64encode(data).decode('utf-8')}"
+        data = response.body
+        if isinstance(event, RoomEncryptedAudio):
+            data = decrypt_attachment(data, event.key["k"], event.hashes["sha256"], event.iv)
```

Comment:
Ignore me if you like. It is not important. 

Wouldn't  a "if-then-else" be better? Easier to read?

Something like:
if isinstance(event, RoomMessageAudio): # unencrypted audio
            data = response.body
elif isinstance(event, RoomEncryptedAudio):
            data = decrypt_attachment(response.body, event.key["k"], event.hashes["sha256"], event.iv)
else:
            logger.error(f"Unsupported event type {event}")
            return

Reply:
> Ignore me if you like. It is not important.
> 
> Wouldn't a "if-then-else" be better? Easier to read?
> 
> Something like: if isinstance(event, RoomMessageAudio): # unencrypted audio data = response.body elif isinstance(event, RoomEncryptedAudio): data = decrypt_attachment(response.body, event.key["k"], event.hashes["sha256"], event.iv) else: logger.error(f"Unsupported event type {event}") return

The last else can't happen - only RoomMessageAudio and RoomEncryptedAudio are registered for this callback. That leaves only two cases - I like my approach better, but I'm not opposed to that approach. Shall I change it?